### PR TITLE
[zh-cn] update & finish translation of 'keyboard navigable javascript widgets'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Related issues and pull requests

- [x] waiting for https://github.com/mdn/content/pull/29903

### CODE REVIEW

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 1]
```diff
@@ -1,105 +1,78 @@
 ---
 title: 键盘导航的 JavaScript 组件
 slug: Web/Accessibility/Keyboard-navigable_JavaScript_widgets
+l10n:
+  sourceCommit: f6d04a43eadf5ab26a3488942dfb318b58234eb5
 ---
 
 <section id="Quick_links">
-  {{ListSubpagesForSidebar("Web/Accessibility", 1)}}
+  {{ListSubpagesForSidebar("/zh-CN/docs/Web/Accessibility", 1)}}
```
Comment (yin1999) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:9]: Hey @jasonren0403. I wonder if we could fix the problem of `ListSubpagesForSidebar` (only displaying en-US titles) so we could list all the articles in the sidebar. There are only a few of them listed (aka, which is translated into Chinese)

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 2]
```diff
@@ -1,105 +1,78 @@
...
 </section>
 
 ### 概况
```
Comment (yin1999) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:13]: Could we remove this (also in en-US)

```suggestion
```

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 3]
```diff
@@ -1,105 +1,78 @@
...
 
-网页应用经常使用 JavaScript 来模拟桌面组件，比如菜单、树状视图、富文本以及标签面板等。网页中这些组件通常是由 {{ HTMLElement("div") }}和 {{ HTMLElement("span") }} 等元素组合而成，并不是原生的，也不会提供对应的桌面组件所拥有的键盘功能。本文档主要描述能让 JavaScript 组件通过键盘 access 的技术。
+网页应用经常使用 JavaScript 来模拟桌面组件，比如菜单、树状视图、富文本以及标签面板等。网页中这些组件通常是由 {{ HTMLElement("div") }}和 {{ HTMLElement("span") }} 等元素组合而成，并不是原生的，也不会提供对应的桌面组件所拥有的键盘功能。本文档主要描述能让 JavaScript 组件使用键盘访问的技术。
 
 ### 利用 tabindex
```
Comment (yin1999) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:16]: Use h2 headings, and the same as the following headings

```suggestion
## 利用 tabindex
```

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 4]
```diff
@@ -145,48 +118,37 @@ _范例 2: 一个使用 tabindex 控制键盘 access 的菜单控件_
 
 #### 方法 1: 流动 tabindex
```
Comment (yin1999) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:119]:
```suggestion
#### 方法 1：流动 tabindex
```

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 5]
```diff
@@ -145,48 +118,37 @@ _范例 2: 一个使用 tabindex 控制键盘 access 的菜单控件_
...
+#### 用 onfocus 来追踪当前焦点
 
-不要认为所有的焦点变化都来自于键盘和鼠标事件：像屏幕阅读器之类的辅助技术可以设置焦点到任意一个可 focus 中的元素上。用 onfocus 和 onblur 来追踪焦点。
+不要认为所有的焦点变化都来自于键盘和鼠标事件：像屏幕阅读器之类的辅助技术可以设置焦点到任意一个可聚焦元素上。用 `onfocus` 和 `onblur` 来追踪焦点。
 
-`onfocus` 和 `onblur` 现在可以在每个元素上使用。没有标准的 DOM 接口来获取当前文档的 focus，所以如果你想追踪就需要保存追踪数据在 JavaScript 变量里。
+`onfocus` 和 `onblur` 现在可以在每个元素上使用。没有标准的 DOM 接口来获取当前文档的焦点，如果要跟踪焦点状态，可以使用 [document.activeElement](/zh-CN/docs/Web/API/Document/activeElement) 来获取活动元素。还可以使用 [document.hasFocus](/zh-CN/docs/Web/API/Document/hasFocus) 来确定当前文档是否处于焦点状态。
 
-#### 方法 2: aria-activedescendant
+#### 方法 2: `aria-activedescendant`
```
Comment (yin1999) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:143]: +1

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md (hunk 1 above)
Comment (yin1999) [reply files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:9]: Try to fix this in mdn/yari#8241. Let's revert the change here

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 6]
```diff
@@ -132,92 +103,77 @@ _范例 2: 一个使用 tabindex 控制键盘 access 的菜单控件_
...
-### 在组合控件里面管理 focus
+## 在组合控件里面管理焦点
 
 当用户从一个组件 tab 离开之后 focus 回来，焦点应该回到离开之时正被 focus 中的元素上，比如某个树节点或者网格单元。有两种办法完成这一点：
 
 1. 流动 `tabindex`: 通过编程移动 focus
 2. `aria-activedescendent`: 管理一个“虚拟”focus
 
-#### 方法 1: 流动 tabindex
+### 方法 1: 流动 tabindex
```
Comment (jasonren0403) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:117]:
```suggestion
### 方法 1：流动 tabindex
```

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 7]
```diff
@@ -132,92 +103,77 @@ _范例 2: 一个使用 tabindex 控制键盘 access 的菜单控件_
...
+#### 用 onfocus 来追踪当前焦点
 
-不要认为所有的焦点变化都来自于键盘和鼠标事件：像屏幕阅读器之类的辅助技术可以设置焦点到任意一个可 focus 中的元素上。用 onfocus 和 onblur 来追踪焦点。
+不要认为所有的焦点变化都来自于键盘和鼠标事件：像屏幕阅读器之类的辅助技术可以设置焦点到任意一个可聚焦元素上。用 `onfocus` 和 `onblur` 来追踪焦点。
 
-`onfocus` 和 `onblur` 现在可以在每个元素上使用。没有标准的 DOM 接口来获取当前文档的 focus，所以如果你想追踪就需要保存追踪数据在 JavaScript 变量里。
+`onfocus` 和 `onblur` 现在可以在每个元素上使用。没有标准的 DOM 接口来获取当前文档的焦点，如果要跟踪焦点状态，可以使用 [document.activeElement](/zh-CN/docs/Web/API/Document/activeElement) 来获取活动元素。还可以使用 [document.hasFocus](/zh-CN/docs/Web/API/Document/hasFocus) 来确定当前文档是否处于焦点状态。
 
-#### 方法 2: aria-activedescendant
+### 方法 2: `aria-activedescendant`
```
Comment (jasonren0403) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:141]:
```suggestion
### 方法 2：`aria-activedescendant`
```

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 8]
```diff
@@ -1,105 +1,76 @@
...
+具有交互功能的原生 HTML 元素，如 {{ HTMLElement("a") }}、{{ HTMLElement("input") }} 和 {{ HTMLElement("select") }}，已经可以被键盘访问，因此使用其中之一是让组件与键盘配合使用的最快途径。
 
-#### 组合控件 Grouping controls
+作者还可以为 {{ HTMLElement("div") }} 或 {{ HTMLElement("span") }} 添加为 `0` 的 `tabindex` 值，使其可以通过键盘访问。这对于使用 HTML 中不存在的交互式元素的组件尤其有用。
 
-针对像菜单、标签面板、树等这些组合控件，父元素应该在 tab 序列里面 (tabindex="0") 而每个后代选择/标签/单元/行应该从 tab 序列里面移除 (tabindex="-1")。用户应该可通过方向键来操控这些后代元素。(关于组件一般预期的键盘支持，可以查看 [DHTML Style Guide](http://access.aol.com/dhtml-style-guide-working-group/).)
+### 组合控件
 
-下面这个组合菜单的例子展示了这种技术的使用。一旦键盘 focus 中容器 ul 元素，Javascript 开发人员需要在键盘事件里针对方向键的响应里管理里面元素的 focus 顺序。在组件里管理 focus 的办法，看下面这个 "在组合控件里管理 focus " 例子。
+对于菜单、选项卡列表、网格或树状视图等分组部件，父元素应处于选项卡顺序中（`tabindex="0"`），而每个子选项/选项卡/单元格/行都应从选项卡顺序中移除（`tabindex="-1"`）。用户应能使用箭头键浏览后代元素。（有关典型窗口小部件通常需要的键盘支持的完整描述，请参阅 [WAI-ARIA 创作实践](https://www.w3.org/TR/wai-aria-practices-1.1/)）。
```
Comment (yin1999) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:71]: mdn/content#30425

```suggestion
对于菜单、选项卡列表、网格或树状视图等分组部件，父元素应处于选项卡顺序中（`tabindex="0"`），而每个子选项/选项卡/单元格/行都应从选项卡顺序中移除（`tabindex="-1"`）。用户应能使用箭头键浏览后代元素。（有关典型窗口小部件通常需要的键盘支持的完整描述，请参阅 [WAI-ARIA 创作实践](https://www.w3.org/WAI/ARIA/apg/)）。
```

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 9]
```diff
@@ -1,105 +1,76 @@
...
         来聚焦该元素。可以在方向键或者其他键的响应里面做。
       </td>
     </tr>
     <tr>
-      <td>Zero (i.e. <code>tabindex="0"</code>)</td>
-      <td>Yes</td>
-      <td>tab 顺序对应于该元素在文档中的位置</td>
+      <td>零（也就是说，<code>tabindex="0"</code>）</td>
+      <td>是</td>
+      <td>相对于元素在文档中的位置，以 tab 顺序排列（请注意，交互式元素如 {{HTMLElement('a')}}} 默认具有这种行为，它们不需要该属性）。</td>
```
Comment (yin1999) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:51]:
```suggestion
      <td>相对于元素在文档中的位置，以 tab 顺序排列（请注意，交互式元素如 {{HTMLElement('a')}} 默认具有这种行为，它们不需要该属性）。</td>
```

File: files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md [hunk 10]
```diff
@@ -1,105 +1,76 @@
 ---
 title: 键盘导航的 JavaScript 组件
```
Comment (yin1999) [inline files/zh-cn/web/accessibility/keyboard-navigable_javascript_widgets/index.md:2]: Let's align the translation:

image

```suggestion
title: 键盘导航的 JavaScript 微件
```

**Review by yin1999 — APPROVED** [review]
LGTM. Thank you @jasonren0403

**Pull request merged**